crates/bevy_audio/src/sinks.rs

```diff
@@ -71,6 +71,10 @@ pub trait AudioSinkPlayback {
 /// Use [`AudioBundle`][crate::AudioBundle] to trigger that to happen.
 ///
 /// You can use this component to modify the playback settings while the audio is playing.
+///
+/// If this component is removed from an entity, and an [`AudioSource`][crate::AudioSource] is
+/// attached to that entity, that [`AudioSource`][crate::AudioSource] will start playing. If
+/// that source is unchanged, that translates to the audio restarting.
```

Comment:
`AudioSink` is created by the audio engine dynamically. This may be a bug, but it's difficult to change, and it's genuinely useful. There's a discussion on the Discord about this in the #audio-dev channel (Reproducer and following discussion: https://canary.discord.com/channels/691052431525675048/749430447326625812/1135716276132642878).

If the `AudioSource` that was used to start the audio hasn't been removed, and the `AudioSink` is removed, it's recreated from the same `AudioSource`, thus being an extremely easy way to restart music. Likewise, if the source is swapped out for another one, this provides a clean way to swap from one song to another instantly.

Reply:
It makes sense that that's what's happening.

That's a personal opinion and a bit out of scope for this PR but this API (if it's intended behavior to restart an audio track or switch track) feels kinda hacky to me.
I would expect:
- either `AudioSink` to be added in `AudioSourceBundle` and `play_queued_audio_system` would query newly-added `AudioSink`, or `play_queued_audio_system` would create `AudioSink` only for newly added `Handle<Source>` and `PlaybackSettings`,
- a system would check for changed `Handle<Source>` to recreate the `AudioSink` if the track changed
- a method on either `AudioSink` or `PlaybackSettings` would allow the track to restart,
- I would also expect that either `PlaybackSettings` is deleted when `AudioSink` is created, or that modifying its fields after would effectively do something

Anyway, for this PR, I think this comment should clearly mention what's happening, something like:
```rust
/// If this component is removed from an entity, and the components from `[`AudioBundle`] are still
/// attached to that entity, a new [`AudioSink`] will be recreated. This will effectively:
/// - reset the playback settings to the values in [`PlaybackSettings`]
/// - restart the audio source from the begining
/// - change the audio source if [`Handle<Source>`](Source) changed
```
(after `You can use this component to modify the playback settings while the audio is playing.`)

Reply:
The idea is not to explain the implementation details but to explain the effect of them; other parts of the documentation don't explain why the engine does what it does with certain things, so why should this?

Also, the more I think about it, the more natural this API feels. It's odd, but not as odd as it could be.